userland/libtock/simple_ble.c

```diff
@@ -80,53 +95,65 @@ int ble_initialize(uint16_t advertising_itv_ms, bool discoverable) {
 }
 
 int ble_start_advertising(void) {
+  int err = allow(BLE_DRIVER_NUMBER, BLE_CFG_GAP_BUF_ALLOW, (void *)adv_data, adv_data_idx);
+  if (err < TOCK_SUCCESS)
+    return err;
+
   return command(BLE_DRIVER_NUMBER, BLE_ADV_START_CMD, 0, 0);
+
 }
 
 int ble_stop_advertising(void) {
   return command(BLE_DRIVER_NUMBER, BLE_ADV_STOP_CMD, 1, 0);
 }
 
 int ble_reset_advertisement(void) {
-  return command(BLE_DRIVER_NUMBER, BLE_ADV_CLEAR_DATA_CMD, 1, 0);
+  int err = command(BLE_DRIVER_NUMBER, BLE_ADV_CLEAR_DATA_CMD, 1, 0);
+  if (err < TOCK_SUCCESS)
+    return err;
+  else {
+    adv_data_idx = 0;
+    memset(adv_data, 0, ADV_DATA_MAX_SIZE);
+    return TOCK_SUCCESS;
+  }
 }
```

Comment:
This could probably be a userland-only thingy and when ```start_advertising/replace_adv_data``` we just replace the buffer!